content/blog/llm-agentic-tool-mesh-empowering-gen-ai-with-retrieval-augmented-generation-rag.md

```diff
@@ -0,0 +1,267 @@
+
+Example usage
+
+```python
+from athon.rag import DataLoader
+
+# Configuration for the Data Loader
+LOADER_CONFIG = {
+    'type': 'ChromaForSentences'
+}
+
+# Initialize the Data Loader
+data_loader = DataLoader.create(LOADER_CONFIG)
+
+# Example collection (retrieved from a DataStorage instance)
+collection = data_storage.get_collection().collection
+
+# List of elements to be inserted
+elements = [
+    {"text": "Generative AI is transforming industries.", "metadata": {"category": "AI", "importance": "high"}},
+    {"text": "This document discusses the impact of AI.", "metadata": {"category": "AI", "importance": "medium"}}
+]
+
+# Insert the elements into the collection
+result = data_loader.insert(collection, elements)
+
+# Handle the insertion result
+if result.status == "success":
+    print("Data successfully inserted into the collection.")
+else:
+    print(f"ERROR:\n{result.error_message}")
+```
+
+## The retrieval process
+
+Once the data has been injected and is ready for use, the retrieval process focuses on fetching the most relevant information based on a given input query. This ensures that the language model has access to the right data to generate accurate and contextually relevant outputs.
+
+1. **Data retrieval**: Use various methods, such as dense or sparse retrieval, to fetch the most relevant data from storage.
+2. **Metadata filtering**:  Apply metadata filters to narrow down search results, ensuring the retrieved data matches the specific needs of the query.
+3. **Chunk Expansion**: Expand the retrieved data chunks to provide comprehensive information for the language model.
+
+![](/img/retrieve.png)
+
+Example usage
+
+```python
+from athon.rag import DataRetriever
+
+# Configuration for the Data Retriever
+RETRIEVER_CONFIG = {
+    'type': 'ChromaForSentences',
+    'expansion_type': 'Section',
+    'sentence_window': 3,
+    'n_results': 10,
+    'include': ['documents', 'metadatas']
+}
+
+# Initialize the Data Retriever
+data_retriever = DataRetriever.create(RETRIEVER_CONFIG)
+
+# Example collection (retrieved from a DataStorage instance)
+collection = data_storage.get_collection().collection
+
+# Query to search within the collection
+query = "What is the impact of Generative AI on industries?"
+
+# Retrieve relevant data based on the query
+result = data_retriever.select(collection, query)
+
+# Handle the retrieval result
+if result.status == "success":
+    for element in result.elements:
+        print(f"TEXT:\n{element['text']}\nMETADATA:\n{element['metadata']}\n")
+else:
+    print(f"ERROR:\n{result.error_message}")
+```
+
+# LLM Agentic Tool Mesh in action: Agentic tool using RAG
+
+In the [LLM Agentic Tool Mesh GitHub](https://github.com/HewlettPackard/llmesh), there is an example of a RAG-based tool that provides quick and accurate access to 5G specifications: the **telco expert**.
```

Comment:
I went to your link here, but couldn't find anything marked "telco expert". I wanted to see if it should read "Telco Expert". But I looked in the Example folder and I looked at the general page it pointed to and didn't see anything that I could identify as such. Would others be able to recognize it?